File: python/abc157/2.py
A1=list(map(int,input().split()))
A2=list(map(int,input().split()))
A3=list(map(int,input().split()))
temp_list = []
temp_list.append(A1)
temp_list.append(A2)
temp_list.append(A3)

bingo_list = [[0 for i in range(3)] for j in range(3)]

N = int(input())
for _ in range(N):
    temp = int(input())
    if (temp in A1):
      bingo_list[0][A1.index(temp)] = 1
    if (temp in A2):
      bingo_list[1][A2.index(temp)] = 1
    if (temp in A3):
      bingo_list[2][A3.index(temp)] = 1

if sum(bingo_list[0]) == 3 or sum(bingo_list[1]) == 3 or sum(bingo_list[2]) == 3 or (bingo_list[0][0]+(bingo_list[1][0])+(bingo_list[2][0])) == 3 or (bingo_list[0][1]+(bingo_list[1][1])+(bingo_list[2][1])) == 3 or (bingo_list[0][2]+(bingo_list[1][2])+(bingo_list[2][2])) == 3:
    print("Yes")
elif (bingo_list[0][0]+(bingo_list[1][1])+(bingo_list[2][2])) == 3 or (bingo_list[0][2]+(bingo_list[1][1])+(bingo_list[2][0])) == 3:
    print("Yes")
else:
    print("No")

File: ts/abc209/2.ts
import * as fs from 'fs';


// input();
const lines: any = fs.readFileSync('/dev/stdin', 'utf8').split(/\n/);

function parseBigInt(line: string): BigInt {
	return BigInt(line)
}

function parseString(line: string): string {
	return line
}

function parseIntArray(line: string): number[] {
	return line.split(' ').map(v => parseInt(v));
}

function parseBigIntArray(line: string): BigInt[] {
	return line.split(' ').map(v => BigInt(v));
}
   
  function parseStrArray(line: string): string[] {
	return line.split('');
}

// output();
const output = (x: any) => console.log(x)

const outputFromArray = (x: any[]) => x.map(x => console.log(x));


function main() {
	// write code
	let [itemNumber, budget] = parseIntArray(lines[0])
	const itemPriceList = parseIntArray(lines[1])

	for (let index = 0; index < itemNumber; index++) {
		if (index % 2 == 1) {
			budget -= (itemPriceList[index]-1);
		} else {
			budget -= itemPriceList[index];
		}
		if (budget < 0) {
			return output('No')
		}
	}
	return output('Yes')
}

// execute main function
main()